lib/pages/modlog_page.dart

```diff
@@ -0,0 +1,519 @@
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class _ModlogEntry {
+  final DateTime when;
+  final UserSafe mod;
+  final Widget action;
+  final String reason;
+
+  const _ModlogEntry({
+    @required this.when,
+    @required this.mod,
+    @required this.action,
+    this.reason,
+  })  : assert(when != null),
+        assert(mod != null),
+        assert(action != null);
+
+  _ModlogEntry.fromModRemovePostView(
+    ModRemovePostView removedPost,
+    Widget action,
+  ) : this(
+          when: removedPost.modRemovePost.when,
+          mod: removedPost.moderator,
+          action: action,
+          reason: removedPost.modRemovePost.reason,
+        );
+
+  _ModlogEntry.fromModLockPostView(
+    ModLockPostView lockedPost,
+    Widget action,
+  ) : this(
+          when: lockedPost.modLockPost.when,
+          mod: lockedPost.moderator,
+          action: action,
+        );
+
+  _ModlogEntry.fromModStickyPostView(
+    ModStickyPostView stickiedPost,
+    Widget action,
+  ) : this(
+          when: stickiedPost.modStickyPost.when,
+          mod: stickiedPost.moderator,
+          action: action,
+        );
+
+  _ModlogEntry.fromModRemoveCommentView(
+    ModRemoveCommentView removedComment,
+    Widget action,
+  ) : this(
+          when: removedComment.modRemoveComment.when,
+          mod: removedComment.moderator,
+          action: action,
+          reason: removedComment.modRemoveComment.reason,
+        );
+
+  _ModlogEntry.fromModRemoveCommunityView(
+    ModRemoveCommunityView removedCommunity,
+    Widget action,
+  ) : this(
+          when: removedCommunity.modRemoveCommunity.when,
+          mod: removedCommunity.moderator,
+          action: action,
+          reason: removedCommunity.modRemoveCommunity.reason,
+        );
+
+  _ModlogEntry.fromModBanFromCommunityView(
+    ModBanFromCommunityView bannedFromCommunity,
+    Widget action,
+  ) : this(
+          when: bannedFromCommunity.modBanFromCommunity.when,
+          mod: bannedFromCommunity.moderator,
+          action: action,
+          reason: bannedFromCommunity.modBanFromCommunity.reason,
+        );
+
+  _ModlogEntry.fromModBanView(
+    ModBanView banned,
+    Widget action,
+  ) : this(
+          when: banned.modBan.when,
+          mod: banned.moderator,
+          action: action,
+          reason: banned.modBan.reason,
+        );
+
+  _ModlogEntry.fromModAddCommunityView(
+    ModAddCommunityView addedToCommunity,
+    Widget action,
+  ) : this(
+          when: addedToCommunity.modAddCommunity.when,
+          mod: addedToCommunity.moderator,
+          action: action,
+        );
+
+  _ModlogEntry.fromModAddView(
+    ModAddView added,
+    Widget action,
+  ) : this(
+          when: added.modAdd.when,
+          mod: added.moderator,
+          action: action,
+        );
+
+  TableRow build(BuildContext context) => TableRow(
+        children: [
+          Center(child: Text(when.fancyShort)),
```

Comment:
make it so that on tap you see the full timestamp


my suggestion:
```
onTap: () {
  showBottomModal(
    context: context,
    builder: (context) => Padding(
    padding: const EdgeInsets.all(8),
      child: Center(child: Text(when.toString())),
    ),
  );
},
```

Reply:
`BottomModal`? I think a dialog fits more here

Reply:
sure